File: files/CH-223.py
import pandas as pd
from datetime import timedelta, time

path = "CH-223 Custom Grouping.xlsx"

input = pd.read_excel(path, usecols="B:D", skiprows=1, nrows=15)
test = pd.read_excel(path, usecols="H:I", skiprows=1, nrows=7).rename(columns=lambda col: col.replace('.1', ''))
input['Date'] = input['Date'].dt.day + (input['Time'] > time(12, 0)).astype(int)
result = input.groupby(input['Date']).Sales.sum().reset_index(name='Sales').assign(Day=lambda df: range(1, len(df) + 1))
result = result[['Day', 'Sales']]
print(result.equals(test)) # True

File: files/200-299/223/CH-223.R
library(tidyverse)
library(readxl)
library(hms)

path = "files/CH-223 Custom Grouping.xlsx"
input = read_excel(path, range = "B2:D16")
test = read_excel(path, range = "H2:I9")

result = input %>%
  mutate(Date = day(Date), Time = hms::as_hms(Time)) %>%
  mutate(Date = ifelse(Time > hms::as_hms("12:00:00"), Date + 1, Date)) %>%
  summarise(Sales = sum(Sales), .by = Date) %>%
  mutate(Day = row_number()) %>%
  select(Day, Sales)

all.equal(result, test, check.attributes = FALSE)
##> [1] TRUE
